cpp/src/arrow/type.h

```diff
@@ -714,9 +714,11 @@ class ARROW_EXPORT FixedSizeBinaryType : public FixedWidthType, public Parametri
 /// \brief Base type class for (fixed-size) decimal data
 class ARROW_EXPORT DecimalType : public FixedSizeBinaryType {
  public:
-  explicit DecimalType(Type::type type_id, int32_t byte_width, int32_t precision,
+  explicit DecimalType(Type::type override_type_id, int32_t byte_width, int32_t precision,
```

Comment:
Is this a reserved word or macro or something in some MSVC environment? If so, we should also add a lint pass which checks for that identifier

Reply:
include\arrow\type.h(710,7): warning C4458:  declaration of 'type_id' hides class member